test/jsupdate-behaviour-mocha.js

```diff
@@ -61,14 +136,40 @@ describe("framework-state", function() {
                 network.graph.addEdge('sink', 'output', 'sut', 'input');
                 test.sendData(network.processes.sink.component, 'input', "tongue");
                 setTimeout(done.bind(this, "nothing happened"), 100);
-            });
+            }).catch(fail);
+        }).should.become("nothing happened");
+    });
+    it("should not fire updater if upstream updater returns undefined after producing an output state", function() {
+        return new Promise(function(done, fail){
+            var count = 0;
+            test.createNetwork({
+                once: jswrapper(function(input) {
+                    switch(++count) {
+                        case 1: return input;
+                        case 2: return;
+                    }
+                }),
+                sut: jswrapper(function(input) {
+                    return "Hello " + input;
+                })
+            }).then(function(network){
+                network.graph.addEdge('once', 'output', 'sut', 'input');
+                return new Promise(function(adv) {
+                    test.onOutPortData(network.processes.sut.component, 'output', adv);
```

Comment:
Maybe add a comment like:
// This is in a promise because we need to wait for the first event to finish propagating
// before registering 'fail' on the output port a few lines below.